From 7d46257741a60291504b998ce0bc5ea1f2e28d00 Mon Sep 17 00:00:00 2001
From: Ignaz Forster <iforster@suse.com>
Date: Tue, 24 Mar 2020 13:41:43 +0100
Subject: [PATCH] Support more architectures
References: gh#coreos/ignition#944
Upstream: submitted

Commit 3b930b made it impossible to build Ignition on any other arch then
s390x, ppc64le, amd64 or arm64 by explicitly stating the allowed platforms.
This made it impossible to build e.g. for 386.
Continue to use the fw_cfg interface as default instead.
---
 internal/providers/qemu/qemu_fwcfg.go | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/internal/providers/qemu/qemu_fwcfg.go b/internal/providers/qemu/qemu_fwcfg.go
index 0758943..c485d6f 100644
--- a/internal/providers/qemu/qemu_fwcfg.go
+++ b/internal/providers/qemu/qemu_fwcfg.go
@@ -12,10 +12,11 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// +build amd64 arm64
+// +build !s390x,!ppc64le
 
-// The QEMU provider on amd64 and arm64 fetches a local configuration from the
-// firmware config interface (opt/com.coreos/config).
+// The default QEMU provider fetches a local configuration from the firmware
+// config interface (opt/com.coreos/config). Platforms without support for
+// qemu_fw_cfg should use the blockdev implementation instead.
 
 package qemu
 
-- 
2.25.1

